ergotree-interpreter/src/eval/spreheader.rs

```diff
@@ -0,0 +1,186 @@
+    }
+
+    #[test]
+    fn test_eval_parent_id() {
+        let expr = create_get_preheader_property_expr(spreheader::PARENT_ID_PROPERTY.clone());
+        let ctx = Rc::new(force_any_val::<Context>());
+        let expected = ctx.pre_header.parent_id.clone();
+        let actual = {
+            let bs = eval_out::<Vec<i8>>(&expr, ctx.clone());
+            block_id_from_bytes_signed(bs)
+        };
+        assert_eq!(expected, actual);
+    }
+
+    #[test]
+    fn test_eval_timestamp() {
+        let expr = create_get_preheader_property_expr(spreheader::TIMESTAMP_PROPERTY.clone());
+        let ctx = Rc::new(force_any_val::<Context>());
+        let expected = ctx.pre_header.timestamp as i64;
+        let actual = eval_out::<i64>(&expr, ctx.clone());
+        assert_eq!(expected, actual);
+    }
+
+    #[test]
+    fn test_eval_n_bits() {
+        let expr = create_get_preheader_property_expr(spreheader::N_BITS_PROPERTY.clone());
+        let ctx = Rc::new(force_any_val::<Context>());
+        let expected = ctx.pre_header.n_bits as i64;
+        let actual = eval_out::<i64>(&expr, ctx.clone());
+        assert_eq!(expected as i64, actual);
+    }
+
+    #[test]
+    fn test_eval_height() {
+        let expr = create_get_preheader_property_expr(spreheader::HEIGHT_PROPERTY.clone());
+        let ctx = Rc::new(force_any_val::<Context>());
+        let expected = ctx.pre_header.height as i32;
+        let actual = eval_out::<i32>(&expr, ctx.clone());
+        assert_eq!(expected, actual);
+    }
+
+    #[test]
+    fn test_eval_miner_pk() {
+        let expr = create_get_preheader_property_expr(spreheader::MINER_PK_PROPERTY.clone());
+        let ctx = Rc::new(force_any_val::<Context>());
+        let expected = ctx.pre_header.miner_pk.clone();
+        let actual = {
+            let pk = eval_out::<EcPoint>(&expr, ctx.clone());
+            Box::new(pk)
+        };
+        assert_eq!(expected, actual);
+    }
+
+    #[test]
+    fn test_eval_votes() {
+        let expr = create_get_preheader_property_expr(spreheader::VOTES_PROPERTY.clone());
+        let ctx = Rc::new(force_any_val::<Context>());
+        let expected = ctx.pre_header.votes.clone();
+        let actual = {
+            let votes_bytes = eval_out::<Vec<i8>>(&expr, ctx.clone()).as_vec_u8();
+            Votes::try_from(votes_bytes)
+                .expect("internal error: votes bytes buffer length isn't equal to 3")
+        };
+        assert_eq!(expected, actual);
+    }
+
+    #[test]
+    fn test_eval_failed_invalid_obj() {
+        // calling for PreHeader property on Context obj
+        let expr: Expr = PropertyCall {
+            obj: Box::new(Expr::Context),
+            method: spreheader::VERSION_PROPERTY.clone(),
+        }
+        .into();
+        assert!(try_eval_out_wo_ctx::<i8>(&expr).is_err());
+    }
+
+    // TODO: [sab] implement after merging https://github.com/ergoplatform/sigma-rust/pull/433
```

Comment:
Let's uncomment it since #433 is merged.

Reply:
done
thank you for the review!